File: backend/app/models/process.py
"""กระบวนการตาม SOP — catalog · การอนุมัติ · checklist · บัญชีซอฟต์แวร์ — 7 ตาราง

ตรงกับ docs/02-data-model.md v2.0 §2.4, §5.10–5.16
ทั้งหมดนี้ SOP-03/04/05/06 บังคับ ไม่ใช่ของที่เลือกทำได้
"""

from __future__ import annotations

from datetime import datetime

from sqlalchemy import (
    BigInteger,
    Boolean,
    CheckConstraint,
    DateTime,
    ForeignKey,
    Index,
    Integer,
    SmallInteger,
    String,
    UniqueConstraint,
)
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.core.constants import (
    ApprovalStatus,
    ApproverType,
    ClockStartEvent,
    Priority,
    TargetMode,
    values,
)
from app.db.base import Base, TimestampMixin


class ServiceCatalogItem(Base, TimestampMixin):
    """เป้าหมายเวลารายรายการของคำขอบริการ (SLA 5.3)

    เหตุผลที่ต้องมีตารางนี้: รีเซ็ตรหัสผ่านมีเป้า 30 นาทีทำการ ขณะที่ P4 คือ
    5 วันทำการ — ต่างกัน 90 เท่า ใช้ sla_target ตาม priority อย่างเดียวแทนไม่ได้
    """

    __tablename__ = "service_catalog_item"
    __table_args__ = (
        UniqueConstraint("company_id", "code"),
        CheckConstraint(f"target_mode IN {tuple(values(TargetMode))}", name="target_mode_valid"),
        CheckConstraint(
            f"clock_start_event IN {tuple(values(ClockStartEvent))}", name="clock_start_valid"
        ),
        CheckConstraint(f"default_priority IN {tuple(values(Priority))}", name="priority_valid"),
        # target_mode = duration ต้องมีจำนวนนาที
        CheckConstraint(
            "target_mode <> 'duration' OR target_minutes IS NOT NULL",
            name="duration_needs_minutes",
        ),
    )

    id: Mapped[int] = mapped_column(BigInteger, primary_key=True)
    company_id: Mapped[int | None] = mapped_column(BigInteger, ForeignKey("company.id"), index=True)
    code: Mapped[str] = mapped_column(String(40), nullable=False)
    name_th: Mapped[str] = mapped_column(String(150), nullable=False)
    category_id: Mapped[int | None] = mapped_column(BigInteger, ForeignKey("ticket_category.id"))
    default_impact: Mapped[str] = mapped_column(String(20), default="individual", nullable=False)
    default_urgency: Mapped[str] = mapped_column(String(20), default="low", nullable=False)
    default_priority: Mapped[str] = mapped_column(
        String(10), default=Priority.P4.value, nullable=False
    )

    target_mode: Mapped[str] = mapped_column(
        String(30), default=TargetMode.DURATION.value, nullable=False
    )
    target_minutes: Mapped[int | None] = mapped_column(Integer)  # นาทีทำการ
    # จุดเริ่มนับต่างกัน 4 แบบ — คำขอสิทธิ์เริ่มนับ "หลังการอนุมัติครบถ้วน" (SLA 5.3)
    clock_start_event: Mapped[str] = mapped_column(
        String(30), default=ClockStartEvent.ON_CREATE.value, nullable=False
    )
    lead_time_days: Mapped[int | None] = mapped_column(Integer)
    lead_time_unit: Mapped[str | None] = mapped_column(String(10))  # calendar / business

    requires_approval: Mapped[bool] = mapped_column(Boolean, default=False, nullable=False)
    # ลำดับ approver_type คั่นด้วย , เช่น "line_manager,system_owner"
    approval_chain: Mapped[str | None] = mapped_column(String(200))
    checklist_template_id: Mapped[int | None] = mapped_column(
        BigInteger, ForeignKey("checklist_template.id")
    )
    is_active: Mapped[bool] = mapped_column(Boolean, default=True, nullable=False)


class ApprovalRequest(Base):
    """การอนุมัติแบบเรียงลำดับ (SOP-03, SOP-06)

    ระหว่างมีขั้น pending ticket จะอยู่ pending_user + pending_reason='approval'
    ซึ่งหยุดนับ SLA ตาม SLA ข้อ 9
    """

    __tablename__ = "approval_request"
    __table_args__ = (
        UniqueConstraint("ticket_id", "seq"),
        CheckConstraint(f"status IN {tuple(values(ApprovalStatus))}", name="status_valid"),
        CheckConstraint(
            f"approver_type IN {tuple(values(ApproverType))}", name="approver_type_valid"
        ),
        # ปฏิเสธต้องมีเหตุผลเสมอ — บังคับที่ระดับฐานข้อมูล
        CheckConstraint(
            "status <> 'rejected' OR comment IS NOT NULL", name="reject_needs_comment"
        ),
        Index("ix_approval_approver_status", "approver_id", "status"),
    )

    id: Mapped[int] = mapped_column(BigInteger, primary_key=True)
    ticket_id: Mapped[int] = mapped_column(
        BigInteger, ForeignKey("ticket.id", ondelete="CASCADE"), nullable=False, index=True
    )
    # ขั้น n+1 เปิดใช้ได้เมื่อขั้น n เป็น approved
    seq: Mapped[int] = mapped_column(SmallInteger, default=1, nullable=False)
    approver_type: Mapped[str] = mapped_column(String(30), nullable=False)
    # null = ยังหาผู้อนุมัติไม่ได้ ต้องแจ้ง company_admin ให้กำหนดคน
    approver_id: Mapped[int | None] = mapped_column(BigInteger, ForeignKey("app_user.id"))
    status: Mapped[str] = mapped_column(
        String(20), default=ApprovalStatus.PENDING.value, nullable=False
    )
    decided_by: Mapped[int | None] = mapped_column(BigInteger, ForeignKey("app_user.id"))
    decided_at: Mapped[datetime | None] = mapped_column(DateTime(timezone=True))
    comment: Mapped[str | None] = mapped_column(String(500))
    requested_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), nullable=False)
    due_at: Mapped[datetime | None] = mapped_column(DateTime(timezone=True))
    # หลักฐานการอนุมัติ (SOP-03 ข้อ 4)
    attachment_id: Mapped[int | None] = mapped_column(BigInteger, ForeignKey("attachment.id"))
    # สิทธิ์ชั่วคราวต้องมีกำหนดสิ้นสุด (SOP-03 ข้อ 6)
    access_expires_at: Mapped[datetime | None] = mapped_column(DateTime(timezone=True))


class ChecklistTemplate(Base, TimestampMixin):
    __tablename__ = "checklist_template"
    __table_args__ = (UniqueConstraint("company_id", "code"),)

    id: Mapped[int] = mapped_column(BigInteger, primary_key=True)
    company_id: Mapped[int | None] = mapped_column(BigInteger, ForeignKey("company.id"))
    code: Mapped[str] = mapped_column(String(40), nullable=False)
    name_th: Mapped[str] = mapped_column(String(150), nullable=False)
    doc_ref: Mapped[str | None] = mapped_column(String(40))  # เช่น AIDC-IT-SOP-001 ก.1
    # เพิ่มทีละ 1 เมื่อแก้รายการ — ticket เก่าอ้างเวอร์ชันเดิมผ่านสแนปช็อต
    version: Mapped[int] = mapped_column(Integer, default=1, nullable=False)
    is_active: Mapped[bool] = mapped_column(Boolean, default=True, nullable=False)

    items: Mapped[list["ChecklistItem"]] = relationship(
        back_populates="template", cascade="all, delete-orphan"
    )


class ChecklistItem(Base):
    __tablename__ = "checklist_item"
    __table_args__ = (UniqueConstraint("template_id", "sort_order"),)

    id: Mapped[int] = mapped_column(BigInteger, primary_key=True)
    template_id: Mapped[int] = mapped_column(
        BigInteger, ForeignKey("checklist_template.id", ondelete="CASCADE"), nullable=False
    )
    sort_order: Mapped[int] = mapped_column(Integer, default=0, nullable=False)
    title_th: Mapped[str] = mapped_column(String(255), nullable=False)
    description: Mapped[str | None] = mapped_column(String(500))
    # true = บล็อกการเปลี่ยนสถานะเป็น resolved (SOP-04 ข้อ 6 / SOP-05 ข้อ 6)
    is_required: Mapped[bool] = mapped_column(Boolean, default=True, nullable=False)
    # true = ต้องแนบไฟล์หลักฐานก่อนติ๊กว่าเสร็จ
    evidence_required: Mapped[bool] = mapped_column(Boolean, default=False, nullable=False)
    default_role_code: Mapped[str | None] = mapped_column(String(30))
    is_active: Mapped[bool] = mapped_column(Boolean, default=True, nullable=False)

    template: Mapped[ChecklistTemplate] = relationship(back_populates="items")


class TicketChecklist(Base):
    __tablename__ = "ticket_checklist"
    __table_args__ = (UniqueConstraint("ticket_id", "template_id"),)

    id: Mapped[int] = mapped_column(BigInteger, primary_key=True)
    ticket_id: Mapped[int] = mapped_column(
        BigInteger, ForeignKey("ticket.id", ondelete="CASCADE"), nullable=False, index=True
    )
    template_id: Mapped[int] = mapped_column(
        BigInteger, ForeignKey("checklist_template.id"), nullable=False
    )
    # สแนปช็อตเวอร์ชัน — แก้ template ภายหลังไม่กระทบ ticket เก่า
    # หลักเดียวกับ ticket.sla_policy_id
    template_version: Mapped[int] = mapped_column(Integer, nullable=False)
    completed_at: Mapped[datetime | None] = mapped_column(DateTime(timezone=True))
    created_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), nullable=False)

    items: Mapped[list["TicketChecklistItem"]] = relationship(
        back_populates="checklist", cascade="all, delete-orphan"
    )


class TicketChecklistItem(Base):
    __tablename__ = "ticket_checklist_item"
    __table_args__ = (
        # ข้อที่ต้องมีหลักฐาน ติ๊กเสร็จไม่ได้ถ้ายังไม่แนบไฟล์
        CheckConstraint(
            "NOT (is_done AND evidence_required AND attachment_id IS NULL)",
            name="evidence_required_when_done",
        ),
    )

    id: Mapped[int] = mapped_column(BigInteger, primary_key=True)
    ticket_checklist_id: Mapped[int] = mapped_column(
        BigInteger, ForeignKey("ticket_checklist.id", ondelete="CASCADE"), nullable=False, index=True
    )
    # null ได้ถ้าต้นทางถูกลบ — ข้อความยังอยู่ใน title_snapshot
    checklist_item_id: Mapped[int | None] = mapped_column(BigInteger, ForeignKey("checklist_item.id"))
    title_snapshot: Mapped[str] = mapped_column(String(255), nullable=False)
    is_required: Mapped[bool] = mapped_column(Boolean, default=True, nullable=False)
    evidence_required: Mapped[bool] = mapped_column(Boolean, default=False, nullable=False)
    is_done: Mapped[bool] = mapped_column(Boolean, default=False, nullable=False)
    done_by: Mapped[int | None] = mapped_column(BigInteger, ForeignKey("app_user.id"))
    done_at: Mapped[datetime | None] = mapped_column(DateTime(timezone=True))
    note: Mapped[str | None] = mapped_column(String(500))
    attachment_id: Mapped[int | None] = mapped_column(BigInteger, ForeignKey("attachment.id"))

    checklist: Mapped[TicketChecklist] = relationship(back_populates="items")


class ApprovedSoftware(Base, TimestampMixin):
    """บัญชีซอฟต์แวร์อนุมัติ (SOP-06 ข้อ 2 / นโยบาย 3.5)

    เฟส 1 เป็น master data อ่านอย่างเดียว — ให้ระบบเช็กได้ว่าคำขอติดตั้ง
    เข้าข่าย SR-SW-INSTALL (2 วันทำการ) หรือ SR-SW-NONSTD (ต้องอนุมัติ)
    """

    __tablename__ = "approved_software"

    id: Mapped[int] = mapped_column(BigInteger, primary_key=True)
    company_id: Mapped[int | None] = mapped_column(BigInteger, ForeignKey("company.id"), index=True)
    name: Mapped[str] = mapped_column(String(150), nullable=False, index=True)
    version: Mapped[str | None] = mapped_column(String(50))
    license_type: Mapped[str | None] = mapped_column(String(50))
    note: Mapped[str | None] = mapped_column(String(500))
    is_active: Mapped[bool] = mapped_column(Boolean, default=True, nullable=False)

File: backend/src/db/schema/process.ts
/**
 * กระบวนการตาม SOP — catalog · การอนุมัติ · checklist · บัญชีซอฟต์แวร์ — 7 ตาราง
 * ตรงกับ docs/02-data-model.md v2.0 §2.4, §5.10–5.16
 *
 * ทั้งหมดนี้ SOP-03/04/05/06 บังคับ ไม่ใช่ของที่เลือกทำได้
 */

import { sql } from 'drizzle-orm';
import {
  bigint,
  bigserial,
  boolean,
  check,
  index,
  integer,
  pgTable,
  smallint,
  timestamp,
  unique,
  varchar,
} from 'drizzle-orm/pg-core';

import {
  APPROVAL_STATUS,
  APPROVER_TYPE,
  CLOCK_START_EVENT,
  PRIORITY,
  TARGET_MODE,
} from '../../common/constants';
import { appUser, company, inList } from './organization';
import { ticket, ticketCategory } from './ticket';

const timestamps = {
  createdAt: timestamp('created_at', { withTimezone: true }).defaultNow().notNull(),
  updatedAt: timestamp('updated_at', { withTimezone: true }).defaultNow().notNull(),
};

export const checklistTemplate = pgTable(
  'checklist_template',
  {
    id: bigserial('id', { mode: 'number' }).primaryKey(),
    companyId: bigint('company_id', { mode: 'number' }).references(() => company.id),
    code: varchar('code', { length: 40 }).notNull(),
    nameTh: varchar('name_th', { length: 150 }).notNull(),
    docRef: varchar('doc_ref', { length: 40 }), // เช่น AIDC-IT-SOP-001 ก.1
    // เพิ่มทีละ 1 เมื่อแก้รายการ — ticket เก่าอ้างเวอร์ชันเดิมผ่านสแนปช็อต
    version: integer('version').default(1).notNull(),
    isActive: boolean('is_active').default(true).notNull(),
    ...timestamps,
  },
  (t) => [unique('uq_checklist_template_company_code').on(t.companyId, t.code)],
);

export const checklistItem = pgTable(
  'checklist_item',
  {
    id: bigserial('id', { mode: 'number' }).primaryKey(),
    templateId: bigint('template_id', { mode: 'number' })
      .notNull()
      .references(() => checklistTemplate.id, { onDelete: 'cascade' }),
    sortOrder: integer('sort_order').default(0).notNull(),
    titleTh: varchar('title_th', { length: 255 }).notNull(),
    description: varchar('description', { length: 500 }),
    // true = บล็อกการเปลี่ยนสถานะเป็น resolved (SOP-04 ข้อ 6 / SOP-05 ข้อ 6)
    isRequired: boolean('is_required').default(true).notNull(),
    // true = ต้องแนบไฟล์หลักฐานก่อนติ๊กว่าเสร็จ
    evidenceRequired: boolean('evidence_required').default(false).notNull(),
    defaultRoleCode: varchar('default_role_code', { length: 30 }),
    isActive: boolean('is_active').default(true).notNull(),
  },
  (t) => [unique('uq_checklist_item_template_order').on(t.templateId, t.sortOrder)],
);

/**
 * เป้าหมายเวลารายรายการของคำขอบริการ (SLA 5.3)
 *
 * เหตุผลที่ต้องมีตารางนี้: รีเซ็ตรหัสผ่านมีเป้า 30 นาทีทำการ ขณะที่ P4 คือ
 * 5 วันทำการ — ต่างกัน 90 เท่า ใช้ sla_target ตาม priority อย่างเดียวแทนไม่ได้
 */
export const serviceCatalogItem = pgTable(
  'service_catalog_item',
  {
    id: bigserial('id', { mode: 'number' }).primaryKey(),
    companyId: bigint('company_id', { mode: 'number' }).references(() => company.id),
    code: varchar('code', { length: 40 }).notNull(),
    nameTh: varchar('name_th', { length: 150 }).notNull(),
    categoryId: bigint('category_id', { mode: 'number' }).references(() => ticketCategory.id),
    defaultImpact: varchar('default_impact', { length: 20 }).default('individual').notNull(),
    defaultUrgency: varchar('default_urgency', { length: 20 }).default('low').notNull(),
    defaultPriority: varchar('default_priority', { length: 10 }).default('P4').notNull(),

    targetMode: varchar('target_mode', { length: 30 }).default('duration').notNull(),
    targetMinutes: integer('target_minutes'), // นาทีทำการ
    // จุดเริ่มนับต่างกัน 4 แบบ — คำขอสิทธิ์เริ่มนับ "หลังการอนุมัติครบถ้วน" (SLA 5.3)
    clockStartEvent: varchar('clock_start_event', { length: 30 }).default('on_create').notNull(),
    leadTimeDays: integer('lead_time_days'),
    leadTimeUnit: varchar('lead_time_unit', { length: 10 }), // calendar / business

    requiresApproval: boolean('requires_approval').default(false).notNull(),
    // ลำดับ approver_type คั่นด้วย , เช่น "line_manager,system_owner"
    approvalChain: varchar('approval_chain', { length: 200 }),
    checklistTemplateId: bigint('checklist_template_id', { mode: 'number' }).references(
      () => checklistTemplate.id,
    ),
    isActive: boolean('is_active').default(true).notNull(),
    ...timestamps,
  },
  (t) => [
    unique('uq_catalog_item_company_code').on(t.companyId, t.code),
    check('ck_catalog_target_mode_valid', inList('target_mode', TARGET_MODE)),
    check('ck_catalog_clock_start_valid', inList('clock_start_event', CLOCK_START_EVENT)),
    check('ck_catalog_priority_valid', inList('default_priority', PRIORITY)),
    // target_mode = duration ต้องมีจำนวนนาที
    check(
      'ck_catalog_duration_needs_minutes',
      sql`target_mode <> 'duration' OR target_minutes IS NOT NULL`,
    ),
    // ต้องอนุมัติ = ต้องระบุว่าใครอนุมัติ
    check(
      'ck_catalog_approval_needs_chain',
      sql`requires_approval = false OR approval_chain IS NOT NULL`,
    ),
    index('ix_catalog_item_company').on(t.companyId),
  ],
);

/**
 * การอนุมัติแบบเรียงลำดับ (SOP-03, SOP-06)
 *
 * ระหว่างมีขั้น pending ticket จะอยู่ pending_user + pending_reason='approval'
 * ซึ่งหยุดนับ SLA ตาม SLA ข้อ 9
 */
export const approvalRequest = pgTable(
  'approval_request',
  {
    id: bigserial('id', { mode: 'number' }).primaryKey(),
    ticketId: bigint('ticket_id', { mode: 'number' })
      .notNull()
      .references(() => ticket.id, { onDelete: 'cascade' }),
    // ขั้น n+1 เปิดใช้ได้เมื่อขั้น n เป็น approved
    seq: smallint('seq').default(1).notNull(),
    approverType: varchar('approver_type', { length: 30 }).notNull(),
    // null = ยังหาผู้อนุมัติไม่ได้ ต้องแจ้ง company_admin ให้กำหนดคน
    approverId: bigint('approver_id', { mode: 'number' }).references(() => appUser.id),
    status: varchar('status', { length: 20 }).default('pending').notNull(),
    decidedBy: bigint('decided_by', { mode: 'number' }).references(() => appUser.id),
    decidedAt: timestamp('decided_at', { withTimezone: true }),
    comment: varchar('comment', { length: 500 }),
    requestedAt: timestamp('requested_at', { withTimezone: true }).defaultNow().notNull(),
    dueAt: timestamp('due_at', { withTimezone: true }),
    // หลักฐานการอนุมัติ (SOP-03 ข้อ 4)
    attachmentId: bigint('attachment_id', { mode: 'number' }),
    // สิทธิ์ชั่วคราวต้องมีกำหนดสิ้นสุด (SOP-03 ข้อ 6)
    accessExpiresAt: timestamp('access_expires_at', { withTimezone: true }),
  },
  (t) => [
    unique('uq_approval_ticket_seq').on(t.ticketId, t.seq),
    check('ck_approval_status_valid', inList('status', APPROVAL_STATUS)),
    check('ck_approval_approver_type_valid', inList('approver_type', APPROVER_TYPE)),
    // ปฏิเสธต้องมีเหตุผลเสมอ — บังคับที่ระดับฐานข้อมูล
    check('ck_approval_reject_needs_comment', sql`status <> 'rejected' OR comment IS NOT NULL`),
    // ตัดสินแล้วต้องมีทั้งคนตัดสินและเวลา
    check(
      'ck_approval_decided_complete',
      sql`status NOT IN ('approved','rejected') OR (decided_by IS NOT NULL AND decided_at IS NOT NULL)`,
    ),
    index('ix_approval_ticket').on(t.ticketId),
    index('ix_approval_approver_status').on(t.approverId, t.status),
  ],
);

export const ticketChecklist = pgTable(
  'ticket_checklist',
  {
    id: bigserial('id', { mode: 'number' }).primaryKey(),
    ticketId: bigint('ticket_id', { mode: 'number' })
      .notNull()
      .references(() => ticket.id, { onDelete: 'cascade' }),
    templateId: bigint('template_id', { mode: 'number' })
      .notNull()
      .references(() => checklistTemplate.id),
    // สแนปช็อตเวอร์ชัน — แก้ template ภายหลังไม่กระทบ ticket เก่า
    // หลักเดียวกับ ticket.sla_policy_id
    templateVersion: integer('template_version').notNull(),
    completedAt: timestamp('completed_at', { withTimezone: true }),
    createdAt: timestamp('created_at', { withTimezone: true }).defaultNow().notNull(),
  },
  (t) => [
    unique('uq_ticket_checklist').on(t.ticketId, t.templateId),
    index('ix_ticket_checklist_ticket').on(t.ticketId),
  ],
);

export const ticketChecklistItem = pgTable(
  'ticket_checklist_item',
  {
    id: bigserial('id', { mode: 'number' }).primaryKey(),
    ticketChecklistId: bigint('ticket_checklist_id', { mode: 'number' })
      .notNull()
      .references(() => ticketChecklist.id, { onDelete: 'cascade' }),
    // null ได้ถ้าต้นทางถูกลบ — ข้อความยังอยู่ใน title_snapshot
    checklistItemId: bigint('checklist_item_id', { mode: 'number' }).references(
      () => checklistItem.id,
    ),
    titleSnapshot: varchar('title_snapshot', { length: 255 }).notNull(),
    isRequired: boolean('is_required').default(true).notNull(),
    evidenceRequired: boolean('evidence_required').default(false).notNull(),
    isDone: boolean('is_done').default(false).notNull(),
    doneBy: bigint('done_by', { mode: 'number' }).references(() => appUser.id),
    doneAt: timestamp('done_at', { withTimezone: true }),
    note: varchar('note', { length: 500 }),
    attachmentId: bigint('attachment_id', { mode: 'number' }),
  },
  (t) => [
    // ข้อที่ต้องมีหลักฐาน ติ๊กเสร็จไม่ได้ถ้ายังไม่แนบไฟล์
    check(
      'ck_checklist_item_evidence_when_done',
      sql`NOT (is_done AND evidence_required AND attachment_id IS NULL)`,
    ),
    // ติ๊กเสร็จแล้วต้องรู้ว่าใครติ๊กและเมื่อไร (เป็นหลักฐานตาม SOP)
    check(
      'ck_checklist_item_done_complete',
      sql`NOT is_done OR (done_by IS NOT NULL AND done_at IS NOT NULL)`,
    ),
    index('ix_ticket_checklist_item_parent').on(t.ticketChecklistId),
  ],
);

/**
 * บัญชีซอฟต์แวร์อนุมัติ (SOP-06 ข้อ 2 / นโยบาย 3.5)
 *
 * เฟส 1 เป็น master data อ่านอย่างเดียว — ให้ระบบเช็กได้ว่าคำขอติดตั้ง
 * เข้าข่าย SR-SW-INSTALL (2 วันทำการ) หรือ SR-SW-NONSTD (ต้องอนุมัติ)
 */
export const approvedSoftware = pgTable(
  'approved_software',
  {
    id: bigserial('id', { mode: 'number' }).primaryKey(),
    companyId: bigint('company_id', { mode: 'number' }).references(() => company.id),
    name: varchar('name', { length: 150 }).notNull(),
    version: varchar('version', { length: 50 }),
    licenseType: varchar('license_type', { length: 50 }),
    note: varchar('note', { length: 500 }),
    isActive: boolean('is_active').default(true).notNull(),
    ...timestamps,
  },
  (t) => [index('ix_approved_software_name').on(t.name), index('ix_approved_software_company').on(t.companyId)],
);
